packages/flutter/lib/src/widgets/heroes.dart

```diff
@@ -115,6 +142,25 @@ class Hero extends StatefulWidget {
   /// {@macro flutter.widgets.child}
   final Widget child;
 
+  /// Optional override to supply a widget that's shown during the hero's flight.
+  ///
+  /// This in-flight widget can depend on the route transition's animation as
+  /// well as the incoming and outgoing routes' [Hero] descendants' widgets and
+  /// layout.
+  ///
+  /// When both the source and destination [Hero]s provide a [flightShuttleBuilder],
+  /// the source's [flightShuttleBuilder] takes precedence.
+  ///
+  /// If none is provided, the destination route's Hero child is shown in-flight
+  /// by default.
+  final HeroFlightShuttleBuilder flightShuttleBuilder;
+
+  /// Placeholder widget left in place as the Hero's child once the flight takes off.
+  ///
+  /// By default, an empty SizedBox keeping the Hero child's original size is
+  /// left in place once the Hero shuttle has taken flight.
+  final TransitionBuilder launchPadBuilder;
```

Comment:
launchPadBuilder is cute but not very intuitive.

We should either:

1. Rename launchPadBuilder and flightShuttleBuilder (e.g. to placeholderBuilder and animationBuilder), or

2. Add an animated diagram to the Hero widget class that shows (in the same style as the AppBar diagram) all the various components of a Hero animation, along with their names.

I don't mind which we do. We'll want to do number 2 eventually anyway. If you like the cute terms and want to keep them, please work with @gspencergoog to make sure we have clear documentation including the aforementioned animated labeled diagram.

Reply:
launchPadBuilder was a bit farcical. I'm keeping flightShuttleBuilder though since there's already so many animations going on (the route animation, the rect tween animation etc) and since we're already using the wording flight, it'd make sense to establish relationship of this particular animation with respect to the flight (i.e. being the thing that's taking the flight).